arthur_bench/client/fs/local_client.py

```diff
@@ -176,112 +155,31 @@ class LocalBenchClient(BenchClient):
-        test_suite_dir = self._test_suite_dir(test_suite_name)
-        if test_suite_dir.is_dir():
-            raise UserValueError(f"test_suite {test_suite_name} already exists")
-        os.mkdir(test_suite_dir)
-        return test_suite_dir
-
-    def _create_run_dir(self, test_suite_name: str, run_name: str) -> Path:
-        run_dir = self._test_suite_dir(test_suite_name) / run_name
-        if os.path.exists(run_dir):
-            raise UserValueError(f"run {run_name} already exists")
-        os.mkdir(run_dir)
-        return run_dir
-
-    def _get_suite_name_from_id(self, id: str) -> Optional[str]:
-        suite_index = json.load(open(self.root_dir / SUITE_INDEX_FILE))
-        if id not in suite_index:
-            return None
-        return suite_index[id]
-
-    def _get_run_name_from_id(self, test_suite_name: str, id: str) -> Optional[str]:
-        run_index = json.load(open(self.root_dir / test_suite_name / RUN_INDEX_FILE))
-        if id not in run_index:
-            return None
-        return run_index[id]
-
-    def _create_test_case_with_id(self, test_case: TestCaseRequest) -> TestCaseResponse:
-        return TestCaseResponse(
-            id=uuid.uuid4(),
-            input=test_case.input,
-            reference_output=test_case.reference_output,
-        )
-
-    def _update_suite_run_time(self, test_suite_name: str, runtime: datetime):
-        suite_file = self.root_dir / test_suite_name / "suite.json"
-        suite = PaginatedTestSuite.parse_file(suite_file)
-        suite.last_run_time = runtime
-        suite.num_runs += 1
-        suite_file.write_text(suite.json())
-
-    def _write_suite_index(self):
-        suite_index_path = self.root_dir / SUITE_INDEX_FILE
-        if suite_index_path.is_file():
-            return
-        json.dump({}, open(suite_index_path, "w"))
-        return None
-
-    def _write_run_index(self, test_suite: str):
-        run_index_path = self.root_dir / test_suite / RUN_INDEX_FILE
-        if run_index_path.is_file():
-            return
-        json.dump({}, open(run_index_path, "w"))
-        return None
-
-    @staticmethod
-    def _update_index(filepath: Path, id: uuid.UUID, name: str):
-        suite_index = json.load(open(filepath))
-        suite_index[str(id)] = name
-        json.dump(suite_index, open(filepath, "w"))
-
-    def _update_suite_index(self, id: uuid.UUID, name: str):
-        suite_index_path = self.root_dir / SUITE_INDEX_FILE
-        LocalBenchClient._update_index(suite_index_path, id, name)
-
-    def _update_run_index(self, test_suite_name: str, id: uuid.UUID, name: str):
-        run_index_path = self.root_dir / test_suite_name / RUN_INDEX_FILE
-        LocalBenchClient._update_index(run_index_path, id, name)
+    def __init__(self, config, use_s3=False):
+        if not use_s3:
+            self.fs_client = LocalFSClient(config)
+        else:
+            raise NotImplementedError("S3 Client not implemented yet")
```

Comment:
probably should think about a better name here than `use_s3` - the intention for now is to write an s3 fs client so we can store results in s3, but I think in the future we'll probably also want to add a client for a nosql db (eg: dynamo)